### PR TITLE
Feat/sorted spike output

### PR DESCRIPTION
**Pull request by ohm314**
Spikes are now sorted before written to file in the output_spikes module. This implementation works both for serial and MPI runs.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -60,6 +60,10 @@ option(ENABLE_CUDA_MODULES "Enable modules like Random123 written in CUDA" ON)
 option(ENABLE_NET_RECEIVE_BUFFERING "Enable event buffering in net_receive function" ON)
 option(ENABLE_OMP_RUNTIME_SCHEDULE "Use runtime schedule for OpenMP" OFF)
 
+## set C++11 standard to be default
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
```
Comment (pramodk) [inline CMakeLists.txt]: Could you create an issue about cxx extensions requirement?
(we should fix that in separate PR)

Comment (ohm314) [reply CMakeLists.txt]: I can, but then we'll have to merge that other PR into master and rebase this branch to not get all kinds of build issues

Comment (pramodk) [reply CMakeLists.txt]: ok thats fine

**Comment by pramodk** [comment]
@ohm314 : could you rebase this on master? And then use `make formatsource` to properly indent the code.

**Pull request merged**